### PR TITLE
Scaling_tests

### PR DESCRIPTION
**Pull request by IvarStefansson**
## Proposed changes

1. Introduces testing for scaling. The functionality is still somewhat experimental, but way better covered than previously. The tests are of integration type, checking that simulations results are identical with and without scaling for all provided model classes. 
2. Fixes some bugs caught by the new tests. 
3. We decided to include the normal gradient in the diffusive interface laws, i.e. division through a/2. Consequently, the normal permeability and thermal conductivity now have the same units as their tangential counterparts (m^2 and W/m/s, respectively).

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply_

- [ ] Minor change (e.g., dependency bumps, broken links, etc).
- [x] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code, etc).
- [ ] Other: 

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.	
- [x] Static typing is included in the update.
- [x] This PR does not duplicated existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).

### CODE REVIEW

File: tests/integration/models/test_energy_balance.py [hunk 1]
```diff
@@ -177,3 +196,46 @@ def test_advection_or_diffusion_dominated(fluid_vals, solid_vals):
...
+    pp.run_time_dependent_model(reference_setup, reference_params)
+
+    params["units"] = pp.Units(**units)
+    setup = EnergyBalanceTailoredBCs(params)
+
+    pp.run_time_dependent_model(setup, params)
+    variable_names = [
+        setup.temperature_variable,
+        setup.pressure_variable,
+        setup.interface_darcy_flux_variable,
```
Comment (keileg) [inline tests/integration/models/test_energy_balance.py:229]: Also interface enthalpy flux?

File: src/porepy/models/material_constants.py [hunk 2]
```diff
@@ -149,6 +154,11 @@ def __init__(self, constants: Optional[dict[str, number]] = None):
             "viscosity": 1,
         }
         if constants is not None:
+            if "s" in constants:
+                if not np.isclose(constants["s"], 1):
+                    raise NotImplementedError(
+                        "Non-unitary time scaling is not implemented."
+                    )
             default_constants.update(constants)
```
[the remark this answers is not in the document]
Comment (IvarStefansson) [reply src/porepy/models/material_constants.py]: You're right, this should be done elsewhere.

**Comment by IvarStefansson** [comment]
> Any reason not to test scaling of temperature? 

Not relevant in the mass balance equation. Could be done for good measure.

File: tests/integration/models/test_mass_balance.py [hunk 3]
```diff
@@ -133,3 +138,45 @@ def test_linear_pressure(fluid_vals, solid_vals):
         k = setup.solid.permeability() / setup.fluid.viscosity()
         grad = 1 / setup.domain_bounds["xmax"]
         assert np.allclose(np.abs(val), normals * grad * k)
+
+
+@pytest.mark.parametrize(
+    "units",
+    [
+        {"m": 2, "kg": 3, "s": 1, "K": 1},
```
[the remark this answers is not in the document]
Comment (IvarStefansson) [reply tests/integration/models/test_mass_balance.py:146]: Not relevant for mass balance. But could be included for good measure. Do you prefer including it?

**Comment by IvarStefansson** [comment]
> Mainly minor comments.
> 
> One high-level suggestion: Should we also slightly expand the tutorial to outline how scaling is done?

Yes, this deserves tutorial documentation. But we might want to wait a few weeks till we've learnt a lesson or two and exposed the functionality a bit more. What do you think?

File: tests/integration/models/test_energy_balance.py (hunk 1 above)
Comment (IvarStefansson) [reply tests/integration/models/test_energy_balance.py:229]: Good idea.

File: tests/integration/models/test_mass_balance.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (keileg) [reply tests/integration/models/test_mass_balance.py:146]: No, let's keep it simple.

**Comment by keileg** [comment]
I agree with regards to the tutorial, let's learn something more first.

**Review by keileg — APPROVED** [review]

**Pull request merged**